test/parser.test.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const { parseDataUnit, parseDataUnitId } = require('../lib/parser');
+
+const {
+  generateRandomUInt8,
+  generateRandomUInt16,
+  generateRandomUInt32,
+} = require('./utils');
+
+test('should parse protocol data unit id', () => {
+  const id = generateRandomUInt8();
+  const buffer = Buffer.from([id]);
+  const parsedId = parseDataUnitId(buffer, 0);
+  expect(parsedId).toStrictEqual(id);
+});
+
+test('should parse handshake', () => {
+  const handshake = {
+    version: generateRandomUInt16(),
+    status: generateRandomUInt8(),
+    token: Buffer.allocUnsafe(32),
+  };
+
+  const buffer = Buffer.allocUnsafe(35);
+  buffer.writeUInt16LE(handshake.version, 0);
+  buffer.writeUInt8(handshake.status, 2);
+  handshake.token.copy(buffer, 3);
+
+  expect(parseDataUnit(buffer, 'HANDSHAKE')).toStrictEqual(handshake);
```

Comment:
I think it's better to have constants storing data unit types ('HANDSHAKE', 'PARCEL_HEADER'...).

Or maybe we can event store schemas in the constants so function call will be like that `parseDataUnit(buffer, PARCEL_HEADER)` and function `parseDataUnit` isn't dependant on the map of all schemas.